### DISCUSSION TITLE
Image uploads

### DISCUSSION THREAD

**Discussion by brucebailey** [MasaCMS/MasaCMS#21] [Q&A]
Does anyone have a video grab or can anyone post a video of how the new drag and drop image upload should be working.  I have tried a few installs now and can't seem to get it functioning properly.

**Comment by chexy** [comment]
I would like to second this.  I cannot seem to get the drag and drop feature to work either.
After dragging the image into the content editor, the image uploads, displays momentarily and then disappears with just the image placeholder box left behind.
The file is actually on the server but the image reference at {siteurl}/index.cfm/_api/asset/image/?filePath={filepath} returns a 404.
We are trying to determine if this is a viable option so we do not have to purchase a ckeditor/ckfinder license.

I have checked for an enable api setting in the site settings area and did not notice anything.  
I have also checked the settings.ini.cfm and found a "suppressapiparams" value set to true.  I set it to false, reloaded the app but that did not seem to change the outcome.

**Reply by jochemd** [reply]
Can you show all your non-default settings in settings.ini.cfm ?

**Reply by chexy** [reply]
Sorry for the delay,
Here is our settings.ini.cfm.  I simply hid some of our specific values by putting a {changed} reference.

[settings]
appreloadkey={changed}
mode=production
[production]
admindir=/admin
admindomain=
adminemail={changed}
adminssl=false
allowautoupdates=true
allowedindexfiles=index.cfm,index.json,index.html
allowlocalfiles=false
alwaysuselocalrenderer=false
assetdir=
assetpath=
autodiscoverplugins=false
autoresetpasswords=false
autoupdateurl=https://codeload.github.com/MasaCMS/MasaCMS/zip/main
bcryptpasswords=true
clientmanagement=false
clientstorage=
confirmsaveasdraft=true
context=
cookiedomain=
cookiepath=
createrequireddirectories=true
customtagpaths=
datasource={changed}
dbpassword=
dbtype=mysql
dbtablespace=USERS
dbusername=
debuggingenabled=true
defaultfilemode=777
defaultthemeurl=https://github.com/MasaCMS/MasaBootstrap4/archive/main.zip
enablemuratag=true
encryptionkey={changed}
errortemplate=/muraWRM/core/templates/error.html
externalconfig=
filedir=
fmshowapplicationroot=true
fmshowsitefiles=true
forceadminssl=true
forcedirectorystructure=true
hashurls=false
haslockablenodes=false
hstsmaxage=1200
imageinterpolation=highQuality
indexfileinurls=false
javaenabled=true
javasettingsloadcoldfusionclasspath=true
javasettingswatchextensions=jar,class
javasettingswatchinterval=60
legacyappcfcsupport=false
legacyjavaloader=false
locale=Server
loginstrikes=4
mailserverip=
mailserverpassword=
mailserverpopport=110
mailserversmtpport=25
mailserverssl=false
mailservertls=false
mailserverusername=
maxarchivedversions=50
maxportalitems=1000
maxsourceimagewidth=4000
mfa=false
mfaperdevice=false
mfasendauthcode=true
notifywithversionlink=true
offline404=true
ormautomanagesession=false
ormcfclocation=
ormdatasource=
ormdbcreate=update
ormenabled=true
ormeventhandling=true
ormflushatrequestend=false
ormlogsql=false
ormsavemapping=false
ormskipcfcwitherror=false
ormusedbformapping=false
ping=true
plugindir=
port=${cgi.SERVER_PORT}
proxypassword=
proxyport=${cgi.SERVER_PORT}
proxyserver=
proxyuser=
purgedrafts=true
recaptchasitekey=
recaptchasecret=
recaptchalanguage=en
rendermuraalerts=true
requesttimeout=1000
sameformfieldsasarray=false
scriptprotect=true
scriptprotectexceptions=body,source,params,objectlist1,objectlist2,objectlist3,objectlist4,objectlist5,objectlist6,objectlist7
securecookies=false
sessioncookiesexpires=never
sendfrommailserverusername=true
sessionhistory=0
sessiontimeout=180
showadminloginhelp=true
siteidinurls=false
sortpermission=editor
strictfactory=true
strongpasswordregex=(?=^.{7,15}$)(?=.*\d)(?![.\n])(?=.*[a-zA-Z]).*$
strongpasswords=false
suppressapiparams=true
tempdir=
testbox=false
title={changed}
tracksessioninnewthread=true
usedefaultsmtpserver=1
usefilemode=false
variations=false
windowdocumentdomain=
databasename=
mysqlengine=InnoDB

**Reply by chexy** [reply]
I forgot to add that this is not a clean Masa install. This is a site which is being upgraded from Mura 7.1.502
